ksqldb-functional-tests/src/test/resources/query-validation-tests/binary-comparison.json

```diff
@@ -182,37 +197,83 @@
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "key": 1, "value": {"KSQL_COL_0": true, "KSQL_COL_1": true, "KSQL_COL_2": true, "KSQL_COL_3": true, "KSQL_COL_4": true, "KSQL_COL_5": true, "KSQL_COL_6": true, "KSQL_COL_7": true, "KSQL_COL_8": true}},
+        {"topic": "OUTPUT", "key": 2, "value": {"KSQL_COL_0": false, "KSQL_COL_1": false, "KSQL_COL_2": false, "KSQL_COL_3": false, "KSQL_COL_4": false, "KSQL_COL_5": false, "KSQL_COL_6": false, "KSQL_COL_7": false, "KSQL_COL_8": false}}
+      ]
+    },
+    {
+      "name": "array comparison",
       "statements": [
         "CREATE STREAM INPUT (A INT KEY, B ARRAY<INT>, C ARRAY<INT>) WITH (kafka_topic='test_topic', value_format='JSON');",
-        "CREATE STREAM OUTPUT AS SELECT A, B = C FROM INPUT;"
+        "CREATE STREAM OUTPUT AS SELECT A, B = C, B <> C, B IS NOT DISTINCT FROM C, B IS DISTINCT FROM C FROM INPUT;"
       ],
-      "expectedException": {
-        "type": "io.confluent.ksql.util.KsqlStatementException",
-        "message": "Cannot compare B (ARRAY<INTEGER>) to C (ARRAY<INTEGER>) with EQUAL."
-      }
+      "inputs": [
+        {"topic": "test_topic", "key": 1, "value": {"B": [1,2], "C": [1,2.0]}},
+        {"topic": "test_topic", "key": 2, "value": {"B": [1,2], "C": [1]}}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "key": 1, "value": {"KSQL_COL_0": true, "KSQL_COL_1": false, "KSQL_COL_2": true, "KSQL_COL_3": false}},
+        {"topic": "OUTPUT", "key": 2, "value": {"KSQL_COL_0": false, "KSQL_COL_1": true, "KSQL_COL_2": false, "KSQL_COL_3": true}}
+      ]
     },
     {
-      "name": "comparison map fails",
+      "name": "map comparison",
       "statements": [
         "CREATE STREAM INPUT (A INT KEY, B MAP<STRING, INT>, C MAP<STRING, INT>) WITH (kafka_topic='test_topic', value_format='JSON');",
-        "CREATE STREAM OUTPUT AS SELECT A, B = C FROM INPUT;"
+        "CREATE STREAM OUTPUT AS SELECT A, B = C, B <> C, B IS NOT DISTINCT FROM C, B IS DISTINCT FROM C FROM INPUT;"
       ],
-      "expectedException": {
-        "type": "io.confluent.ksql.util.KsqlStatementException",
-        "message": "Cannot compare B (MAP<STRING, INTEGER>) to C (MAP<STRING, INTEGER>) with EQUAL."
-      }
+      "inputs": [
+        {"topic": "test_topic", "key": 1, "value": {"B": {"a": 1}, "C": {"a": 1.0}}},
+        {"topic": "test_topic", "key": 2, "value": {"B": {"a": 1}, "C": {"a": 2}}},
+        {"topic": "test_topic", "key": 3, "value": {"B": {"a": 1}, "C": {"b": 1}}}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "key": 1, "value": {"KSQL_COL_0": true, "KSQL_COL_1": false, "KSQL_COL_2": true, "KSQL_COL_3": false}},
+        {"topic": "OUTPUT", "key": 2, "value": {"KSQL_COL_0": false, "KSQL_COL_1": true, "KSQL_COL_2": false, "KSQL_COL_3": true}},
+        {"topic": "OUTPUT", "key": 3, "value": {"KSQL_COL_0": false, "KSQL_COL_1": true, "KSQL_COL_2": false, "KSQL_COL_3": true}}
+      ]
     },
     {
-      "name": "comparison struct fails",
```

Comment:
we should still check that comparisons fail that should fail (e.g. `>` and `<`)

Reply:
Restored